### PR TITLE
Platforms

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request
Adds Platforms
Sprites by Spooky. 
image

## Why It's Good For The Game
Good mapping tool/trick. Helps create some of those 3d-ish spaces

## Changelog

:cl:
add: Platforms. Please use them in maps. they look cool. 
imageadd: Spooky: Platform sprites

/:cl:

### CODE REVIEW

**Comment by Imaginos16** [comment]
image

Here's a mockup by Spooky of idealized platform usage and implementation!

File: code/game/objects/structures/platforms.dm [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+/obj/structure/platform/proc/on_exit(datum/source, atom/movable/leaving, direction)
+	SIGNAL_HANDLER
+
+	if(leaving == src)
+		return // Let's not block ourselves.
+
+	if(!(direction & dir))
+		return
+
+	if (!density)
```
Comment (thgvr) [inline code/game/objects/structures/platforms.dm]:
```suggestion
	if(!density)
```
you.

Comment (Erikafox) [reply code/game/objects/structures/platforms.dm]: look i just copied railing code

**Review by Imaginos16 — APPROVED** [review]
Sprice is good

**Comment by Erikafox** [comment]
i wanna map with these baddies....

**Comment by thgvr** [comment]
fix all the spaced `if`s and ill merge this

File: code/game/objects/structures/platforms.dm [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+				to_chat(user, span_notice("You repair [src]."))
+		else
+			to_chat(user, span_warning("[src] is already in good condition!"))
+		return
+
+/obj/structure/platform/deconstruct_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(!I.tool_start_check(user, amount=0))
+		return FALSE
+	if (I.use_tool(src, user, 3 SECONDS, volume=0))
```
Comment (thgvr) [inline code/game/objects/structures/platforms.dm]:
```suggestion
	if(I.use_tool(src, user, 3 SECONDS, volume=0))
```
"fixed"

**Pull request merged**